mysql-test/suite/innodb/t/stat_tables.test

```diff
@@ -15,3 +15,16 @@ select * from t1;
 drop table t1;
 rename table mysql.table_stats_save to mysql.table_stats;
 flush tables;
+
+--echo #
+--echo # MDEV-32667 dict_stats_save_index_stat() reads uninitialized
+--echo #              index->stats_error_printed
+--echo #
+call mtr.add_suppression("InnoDB: Cannot save index statistics for table");
+CREATE TABLE t (a INT) ENGINE=InnoDB STATS_PERSISTENT=1 STATS_AUTO_RECALC=1;
+BEGIN;
+SELECT COUNT(*)>=0 FROM mysql.innodb_index_stats LOCK IN SHARE MODE;
+INSERT INTO t VALUES(0),(0);
+SELECT sleep(5);
+COMMIT;
```

Comment:
I’m not a friend of `sleep` in regression tests, but I don’t think that there is a better way of reproducing this bug in 10.5. With the minimum `innodb_lock_wait_timeout=1`, the complex InnoDB timeout handling mechanism before 10.6 makes the minimum timeout more like 2 seconds, and we need some safety margin as well. Luckily, if the delay is too short for a slow builder, the test will not fail, but it may merely catch the bug, should it reappear.